src/d/actor/d_a_wall.cpp

```diff
@@ -4,99 +4,354 @@
+    /* a */ 0xC8,
+};
+
+static dCcD_SrcTri l_tri_src = {
+    // dCcD_SrcGObjInf
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0,
+        /* SrcObjAt  Atp     */ 0,
+        /* SrcObjAt  SPrm    */ 0,
+        /* SrcObjTg  Type    */ AT_TYPE_BOMB,
+        /* SrcObjTg  SPrm    */ G_TG_SPRM_SHIELD | G_TG_SPRM_SHIELD_FRONT_RANGE,
+        /* SrcObjCo  SPrm    */ 0,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ G_TG_SPRM_NO_HIT_MARK,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGTriS
+    {
+        /* a */ 0.0f,
+        0.0f,
+        0.0f,
+        /* b */ 0.0f,
+        0.0f,
+        0.0f,
+        /* c */ 0.0f,
+        0.0f,
+        0.0f,
+    },
+};
+
+daWall_c::daWall_c() : mSmokeCb(m_smoke_color, NULL, 0) {}
 
-/* 00000078-00000100       .text _delete__8daWall_cFv */
 BOOL daWall_c::_delete() {
-    /* Nonmatching */
+    mSmokeCb.end();
+    if (heap != NULL && mState == false)
+        dComIfG_Bgsp()->Release(mpBgW);
+
+    dComIfG_resDelete(&mPhs, m_arcname[mType]);
+    return TRUE;
 }
 
-/* 00000100-00000120       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daWall_c*)i_this)->CreateHeap();
 }
 
 /* 00000120-00000288       .text CreateHeap__8daWall_cFv */
-void daWall_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daWall_c::CreateHeap() {
+    J3DModelData* modelData =
+        (J3DModelData*)(dComIfG_getObjectRes(m_arcname[mType], m_bmdname[mType]));
+    JUT_ASSERT(0x181, modelData != 0);
+
+    mpModel = mDoExt_J3DModel__create(modelData, 0x80000, 0x11000022);
+    if (mpModel == NULL)
+        return FALSE;
+
+    setMoveBGMtx();
+    mpBgW = new dBgW();
+
+    if (mpBgW) {
+        bool error = mpBgW->Set((cBgD_t*)dComIfG_getObjectRes(m_arcname[mType], m_dzbname[mType]),
+                                cBgW::MOVE_BG_e, &mMtx);
+        return error != true ? TRUE : FALSE;
+    }
+
+    return false;
 }
 
 /* 00000288-00000380       .text CreateInit__8daWall_cFv */
 void daWall_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    Vec cullMin = m_cull_size[mType * 2];
+    Vec cullMax = m_cull_size[mType * 2 + 1];
+    fopAcM_setCullSizeBox(this, cullMin.x, cullMin.y, cullMin.z, cullMax.x, cullMax.y, cullMax.z);
+    fopAcM_setCullSizeFar(this, 1.0f);
+    mStts.Init(0xFF, 0xFF, this);
+    set_tri();
+    dComIfG_Bgsp()->Regist(mpBgW, this);
+    set_mtx();
+    mpBgW->Move();
+    mSwitchIndex = fopAcM_GetParam(this) & 0xFF;
+    mState = false;
 }
 
-/* 00000380-000004EC       .text _create__8daWall_cFv */
```

Comment:
I'd recommend keeping these comments in, they don't hurt anything and can be good to have as reference sometimes